Sources/PointFree/RequestContext.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import HttpPipeline
+import Prelude
+
+/// A value that wraps any given data with additional context that is useful for completing the
+/// request-to-response lifecycle.
+struct RequestContext<A> {
+  private(set) var currentUser: Database.User? = nil
+  private(set) var currentRequest: URLRequest
```

Comment:
Maybe the request context should carry around the current route instead of bare request. not sure if there might be a situation in which you need the actual request...

Reply:
🤔 Seems like maybe the route would be tougher to _use_?